pkg/util/kubernetes/apiserver/services.go

```diff
@@ -89,13 +92,54 @@ func (m ServicesMapper) Map(nodeName string, pods v1.PodList, endpointList v1.En
 	return nil
 }
 
+// mapOnRef matches pods to services via endpoint TargetRef objects. It supports Kubernetes 1.3+
+func (m ServicesMapper) mapOnRef(nodeName string, endpointList v1.EndpointsList) error {
+	uidToPod := make(map[types.UID]v1.ObjectReference)
+	uidToServices := make(map[types.UID][]string)
+
+	for _, svc := range endpointList.Items {
+		for _, endpointsSubsets := range svc.Subsets {
+			for _, edpt := range endpointsSubsets.Addresses {
+				if edpt.TargetRef == nil {
+					log.Debug("Empty TargetRef on endpoint %s of service %s, skipping", edpt.IP, svc.Name)
+					continue
+				}
+				ref := *edpt.TargetRef
+				if ref.Kind != "Pod" {
+					continue
+				}
+				if ref.Name == "" || ref.Namespace == "" {
```

Comment:
Out of curiosity, does that ever happen (with non-pod objects already excluded)? If so in which case? It would be nice if we can make this log line more informative.